### PR TITLE
Closes Issue #2374: Include more details about issues and pull requests

### LINKED ISSUE #2374: Include more details about Issue and Pull Requests in title attribute
In #2367 we're adding GitHub info do a post.  We show Issues and Pull Requests discussed in a post as numbers only.  This is good for the space we have available, but when we hover over them, it would be good to add more detail.

For example, if a post talks about 2 or more repos, and then shows Issues and Pull Requests, which repo do they go with?  Instead of using `Issue #...` in the `title` attribute on hover, let's also include the repo details (user/org + repo name): `user/repo Issue#...`.  Same for Pull Requests and [Commits](https://github.com/Seneca-CDOT/telescope/issues/2372).

### PR DESCRIPTION
**Pull request by HyperTHD**
## Issue This PR Addresses:
This PR closes #2374 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [x] **UI**: Change which improves UI

## Description

This PR adds additional information to indicate to the user where an issue or pull request comes from. Sentences are unoriginal, but the user will now know what repo an issue or PR comes from when hovering over the issue/PR number. 
This will also include the user as well

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Review by humphd — DISMISSED** [review]
@HyperTHD nice.  Do you want to merge this now, or add the same for Commits in this PR?  Happy either way, but let's file an issue for follow-up if we don't do it here.

**Comment by HyperTHD** [comment]
> @HyperTHD nice. Do you want to merge this now, or add the same for Commits in this PR? Happy either way, but let's file an issue for follow-up if we don't do it here.

It's a simple add-on. I'll rebase and quickly add that in for the commits. Once that's done, you can merge this in

**Comment by HyperTHD** [comment]
> That's an elegant way to extract the info, wonder if I can do the same for filtering urls.

I've made a hook with filtering search results before so it's definitely possible. Also, I find regex to be very confusing at times so if I can ever avoid using it, it's worth it. I only use it if absolutely necessary

**Comment by Andrewnt219** [comment]
@HyperTHD if it works with all the cases in the comments, then let file an issue to change it. Probably have to write a test suite for that as well.

File: src/web/src/components/Posts/Commits.tsx [hunk 1]
```diff
@@ -61,7 +66,7 @@ const Commits = ({ commitUrls }: Props) => {
               href={`https://github.com${url}`}
               rel="bookmark"
               target="_blank"
-              title={'Commit ' + getCommitNumber(url)}
+              title={`${getCommitInfo(url)} Commit Hash ${getCommitNumber(url)}`}
```
Comment (humphd) [inline src/web/src/components/Posts/Commits.tsx:69]: Let's drop the `Hash `, the rest is great.

**Review by humphd — CHANGES_REQUESTED** [review]
Tested on staging + Vercel, works great!  One small fix and this is R+ from me.

File: src/web/src/components/Posts/Commits.tsx (hunk 1 above)
Comment (HyperTHD) [reply src/web/src/components/Posts/Commits.tsx]: Ready to merge!

**Review by humphd — APPROVED** [review]

**Comment by humphd** [comment]
@HyperTHD I'm fine for this to go in as is.  We're low on reviewers atm.

**Review by PedroFonsecaDEV — APPROVED** [review]
LGTM!

**Pull request merged**